src/cryojax/simulator/_integrators/_nufft_project.py

```diff
@@ -29,35 +29,35 @@ class NufftProject(AbstractPotentialIntegrator, strict=True):
         See ``jax-finufft`` for documentation.
     """
 
-    config: ImageConfig
-
     eps: float = field(static=True, default=1e-6)
 
     def __call__(
-        self, potential: RealVoxelGridPotential | RealVoxelCloudPotential
+        self,
+        potential: RealVoxelGridPotential | RealVoxelCloudPotential,
+        config: ImageConfig,
     ) -> ComplexImage:
         """Rasterize image with non-uniform FFTs."""
         if isinstance(potential, RealVoxelGridPotential):
             shape = potential.shape
             fourier_projection = project_with_nufft(
                 potential.real_voxel_grid.ravel(),
                 potential.wrapped_coordinate_grid.get().reshape((math.prod(shape), 3)),
-                self.config.padded_shape,
+                config.padded_shape,
```

Comment:
I suppose this makes things more functional programming. less states and params stored in self

Reply:
its more about the right hierarchy of the pytree I think. the `ImageConfig` needed to be much more privileged because so many objects need it to be passed to them. not sure if this relates to the functional programming aspect of things... i think of this more in the way in which `eqx.Module`s are passed around and operated on by things like `eqx.tree_at` and `jax.tree_util.tree_map`